crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -155,81 +155,66 @@ impl AuthorityStorePruner {
-            // Skipping because  checkpoint's epoch or checkpoint number is too new.
-            // We have to respect the highest executed checkpoint watermark because there might be
-            // parts of the system that still require access to old object versions (i.e. state accumulator)
-            if (current_epoch < checkpoint.epoch() + config.num_epochs_to_retain)
-                || (checkpoint_number > highest_executed_checkpoint)
-            {
-                break;
-            }
-            checkpoints_in_batch += 1;
-            if network_total_transactions == checkpoint.network_total_transactions {
-                continue;
-            }
-            network_total_transactions = checkpoint.network_total_transactions;
 
-            let content = checkpoint_store
-                .get_checkpoint_contents(&checkpoint.content_digest)?
-                .ok_or_else(|| anyhow::anyhow!("checkpoint content data is missing"))?;
+        loop {
+            let checkpoints: Vec<_> = checkpoint_store
+                .certified_checkpoints
+                .iter()
+                .skip_to(&(checkpoint_number + 1))?
+                .take(config.max_checkpoints_in_batch)
+                .map(|(_, ckpt)| ckpt.into_inner())
+                // Filtering out transactions whose epoch or checkpoint number is too new.
+                // We have to respect the highest executed checkpoint watermark because there might be
+                // parts of the system that still require access to old object versions (i.e. state accumulator)
+                .filter(|checkpoint| {
+                    checkpoint.epoch() + config.num_epochs_to_retain <= current_epoch
+                        && *checkpoint.sequence_number() <= highest_executed_checkpoint
+                })
+                .collect();
+
+            checkpoint_number = checkpoints
+                .last()
+                .map(|c| *c.sequence_number())
+                .unwrap_or(checkpoint_number);
+            let is_last_iteration = checkpoints.len() < config.max_checkpoints_in_batch;
+
+            let digests: Vec<_> = checkpoints.into_iter().map(|c| c.content_digest).collect();
+            let execution_digests = checkpoint_store
+                .multi_get_checkpoint_content(&digests)?
+                .into_iter()
+                .flat_map(|contents| {
+                    contents
+                        .expect("checkpoint content is missing")
+                        .into_inner()
+                })
+                .map(|tx| tx.effects);
+
             let effects = perpetual_db
                 .effects
-                .multi_get(content.iter().map(|tx| tx.effects))?;
-
-            if effects.iter().any(|effect| effect.is_none()) {
-                return Err(anyhow::anyhow!("transaction effects data is missing"));
-            }
-            batch_effects.extend(effects.into_iter().flatten());
+                .multi_get(execution_digests)?
```

Comment:
now we are performing a multi-get here for a bunch of checkpoints. The current value for the checkpoint batch in most configurations is 200. Assuming 1k tps, that's 200k in a single multi-get. This could potentially cause issues with RocksDB. 
Should we decrease the batch size in our configurations to a lower value? cc @sadhansood